demo-app/src/main/java/io/opentelemetry/android/demo/clients/RecommendationService.kt

```diff
@@ -0,0 +1,29 @@
+package io.opentelemetry.android.demo.clients
+
+import io.opentelemetry.android.demo.model.Product
+import io.opentelemetry.android.demo.ui.shop.cart.CartViewModel
+
+class RecommendationService(
+    private val productCatalogClient: ProductCatalogClient,
+    private val cartViewModel: CartViewModel
+) {
+
+    fun getRecommendedProducts(currentProduct: Product): List<Product> {
+        val allProducts = productCatalogClient.get()
+        val cartItems = cartViewModel.cartItems.value
+
+        return allProducts.filter { product ->
+            product.id != currentProduct.id && cartItems.none { it.product.id == product.id }
+        }.shuffled().take(4)
```

Comment:
This 4 is kinda hidden and makes the reader wonder how many might be returned. What do you think about making it a parameter with a default value?